hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/HandleContext.java

```diff
@@ -58,4 +63,29 @@ public interface HandleContext {
      * @return the validator for expiry metadata
      */
     ExpiryValidator expiryValidator();
+
+    /**
+     * Gets the {@link SignatureVerification} for the given key. If this key was not provided during
+     * pre-handle, then there will be no corresponding {@link SignatureVerification}. If the key was provided during
+     * pre-handle, then the corresponding {@link SignatureVerification} will be returned with the result of that
+     * verification operation.
+     *
+     * @param key the key to get the verification for
+     * @return the verification for the given key, or {@code null} if no such key was provided during pre-handle
+     */
+    @Nullable
+    SignatureVerification verificationFor(@NonNull final Key key);
+
```

Comment:
Question for @tinker-michaelj : Do we ever need the ability, from the transaction handlers, to get the verification status for a single key? Or do we just want either the total pass/fail information or the Key for a given hollow account?

Reply:
The schedule service and (for at least some time) the contract service will both need the list of cryptographic keys with valid signatures. 

But I can't immediately think of a case where handler needs to re-check if a Hedera `Key` has an active signature.

Reply:
@tinker-michaelj and I discussed this question this morning, and we concluded that the method signature is OK (we don't need to be able to get a list of keys), but the specification is wrong. If somebody passes a cryptographic key (i.e. a primitive `Key` such as ed25519 / ecdsa / etc) then we should get back the `SignatureVerification` for that key. But if they pass in a KeyList or ThresholdKey (or a whole hierarchy of them), then we should also return a `SignatureVerification` that represents pass/fail for that whole structure.